sale_loyalty_incompatibility/tests/test_sale_coupon_incompatibility.py

```diff
@@ -1,99 +1,82 @@
 # Copyright 2023 Tecnativa - Stefan Ungureanu
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 from odoo.exceptions import UserError
-from odoo.tests import Form
 
-from odoo.addons.coupon_incompatibility.tests.test_coupon_incompatibility import (
-    CouponIncompatibilityCase,
-)
+from odoo.addons.sale_loyalty.tests.common import TestSaleCouponCommon
 
 
-class TestCouponIncompatibility(CouponIncompatibilityCase):
-    def _create_coupon_program(self, product=False, code=False, apply_on_order=True):
-        """Helper method to create coupon programs in the tests cases"""
-        coupon_program_form = Form(
-            self.env["coupon.program"],
-            view="sale_coupon.sale_coupon_program_view_promo_program_form",
-        )
-        coupon_program_form.name = "Test Coupon Incompatibility"
-        if product:
-            # We don't want demo programs spoiling our tests
-            coupon_program_form.rule_products_domain = "[('id', '=', %s)]" % (
-                product.id
-            )
-        if not code:
-            coupon_program_form.promo_code_usage = "no_code_needed"
-        else:
-            coupon_program_form.promo_code_usage = "code_needed"
-            coupon_program_form.promo_code = code
-        if apply_on_order:
-            coupon_program_form.promo_applicability = "on_current_order"
-        else:
-            coupon_program_form.promo_applicability = "next_order"
-        coupon_program_form.reward_type = "product"
-        coupon_program_form.reward_product_id = product or self.product_a
-        return coupon_program_form.save()
-
-    def _generate_coupons(self, program, number=1):
-        """Helper method to easily generate coupons in the test cases"""
-        self.env["coupon.generate.wizard"].with_context(active_id=program.id).create(
-            {"generation_type": "nbr_coupon", "nbr_coupons": number}
-        ).generate_coupon()
-        return program.coupon_ids
-
-    def _create_sale(self, partner, products=False):
-        """Helper method to create sales in the test cases"""
-        sale_form = Form(self.env["sale.order"])
-        sale_form.partner_id = partner
-        if not products:
-            products = self.product_a
-        for product in products:
-            with sale_form.order_line.new() as line_form:
-                line_form.product_id = product
-                line_form.product_uom_qty = 3
-        return sale_form.save()
-
-    def _apply_coupon(self, order, code, bp=False):
-        """Helper method to apply either coupon or progam codes. It ensures that the
-        UserError exception is raised as well."""
-        self.env["sale.coupon.apply.code"].with_context(
-            active_id=order.id, bp=bp
-        ).create({"coupon_code": code}).process_coupon()
-
+class TestLoyaltyIncompatibility(TestSaleCouponCommon):
```

Comment:
It's ok using a commonground  case to simplify our test code, but try documenting well what we're going to test and the expected results.